packages/navigator/src/browser/navigator-contribution.ts

```diff
@@ -5,13 +5,50 @@
  * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  */
 
-import { injectable } from "inversify";
+import { injectable, postConstruct, inject } from "inversify";
+import { Widget, FocusTracker } from "@phosphor/widgets";
 import { FILE_NAVIGATOR_ID, FileNavigatorWidget } from './navigator-widget';
 import { AbstractViewContribution } from '@theia/core/lib/browser/shell/view-contribution';
+import URI from '@theia/core/lib/common/uri';
+import { ISelectableTreeNode } from '@theia/core/lib/browser';
+import { MenuModelRegistry } from '@theia/core/lib/common/menu';
+import { CommandRegistry } from '@theia/core/lib/common/command';
+import { ApplicationShell } from '@theia/core/lib/browser/shell/application-shell';
+import { SHELL_TABBAR_CONTEXT_MENU } from '@theia/core/lib/browser/shell/tab-bars';
+import { NavigatorPreferences } from './navigator-preferences';
+
+export namespace FileNavigator {
+
+    export type Revealable = { readonly uri: URI } | URI;
```

Comment:
to discuss: Could we use `SelectionService` and `UriSelection` instead?

Reply:
I don't think it is the same. What we are looking for is a widget that has been opened for a file and is active.